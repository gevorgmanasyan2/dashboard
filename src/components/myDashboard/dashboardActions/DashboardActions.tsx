import React, { FC, MouseEvent, ReactNode, useState, useEffect } from 'react'
import { PlusOutlined } from '@ant-design/icons'
import { Link } from 'react-router-dom'
import fetch from 'node-fetch'
import { connect } from 'react-redux'
import IsLoading from '../../../spinner/IsLoading'
import { config, baseAPIURL } from '../../../api/constants'

import './dashboardActions.scss'

interface DashboardActionsProps {
  icon: ReactNode | null
  title: string
  count: number
  isLoading: boolean
  itemTitle: string
  itemInfo: string
  loc?: LocationType
  apiEndpoint: string
  onClick?: (event: MouseEvent<HTMLElement, MouseEvent>) => void
}
type LocationType = {
  link: string
  addLink?: string
}

type fetchResponse = any

const DashboardActions: FC<DashboardActionsProps> = ({
  icon,
  title,
  itemTitle,
  itemInfo,
  isLoading,
  apiEndpoint,
  loc,
}) => {
  const [clicked, setClicked] = useState<boolean>(false)
  const [totalCount, setTotalCount] = useState<number>(0)
  useEffect(() => {
    const fetchCount = async () => {
      const totalCountResponse: fetchResponse = await fetch(`${baseAPIURL}${apiEndpoint}`, config())
      const totalCount = await totalCountResponse.json()
      setTotalCount(totalCount.count)
    }
    fetchCount()
  }, [apiEndpoint])

  return (
    <>
      <section
        className={
          !clicked ? 'dashboardActions__container' : 'dashboardActions__container--selected'
        }
        onClick={() => setClicked(!clicked)}
      >
        <div className="dashboardActions__container__header">
          <div className="dashboardActions__container__header--icon">{icon}</div>
          <p className="dashboardActions__container__header--title">{title}</p>
        </div>
        <div className="dashboardActions__container__main">
          <div className="dashboardActions__container__main__element">
            <span className="dashboardActions__container__main__element--count">
              <Link to={loc?.link || ''}>{isLoading ? <IsLoading /> : totalCount}</Link>
            </span>
            <span className="dashboardActions__container__main__element--title"> {itemTitle}</span>
          </div>
          <div className="dashboardActions__container__main__icon">
            <Link to={loc?.addLink || ''}>
              <PlusOutlined />
            </Link>
          </div>
        </div>
        <div className="dashboardActions__container__info">
          <span>{itemInfo}</span>
        </div>
      </section>
    </>
  )
}

const mapStateToProps = (state: any) => {
  return {
    isLoading: state.waiver.isLoading,
  }
}

export default connect(mapStateToProps)(DashboardActions)
