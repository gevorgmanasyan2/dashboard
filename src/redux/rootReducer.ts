import { combineReducers } from 'redux'
import * as auth from '../redux/auth/authReducer'
import * as data from './count-data/fetchDataReducer'
import * as waivers from './waiver/fetchWaiverDataReducer'
import * as location from './location/locationReducer'
import * as messages from './messages/messagesReducer'
import * as user from '../redux/my-account/userReducer'
import { authReducer } from './auth/authReducer'
import { fetchDataReducer } from './count-data/fetchDataReducer'
import { fetchWaiverDataReducer } from './waiver/fetchWaiverDataReducer'
import { locationReducer } from './location/locationReducer'
import { messagesReducer } from './messages/messagesReducer'
import { userReducer } from './my-account/userReducer'
export interface CombinedState {
  auth: auth.State
  data: data.State
  waivers: waivers.State
  location: location.State
  messages: messages.State
  user: user.State
}
const rootReducer = combineReducers({
  auth: authReducer,
  data: fetchDataReducer,
  waiver: fetchWaiverDataReducer,
  location: locationReducer,
  messages: messagesReducer,
  user: userReducer,
})

export type RootState = ReturnType<typeof rootReducer>
export default rootReducer
