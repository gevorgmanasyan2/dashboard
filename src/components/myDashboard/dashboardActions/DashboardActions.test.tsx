/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import DashboardActions from './DashboardActions'
import { RightCircleOutlined } from '@ant-design/icons'
import { Provider } from 'react-redux'
import { store } from '../../../redux/store'

it('should render Regulatory component', () => {
  const mockItem = {
    icon: <RightCircleOutlined />,
    title: 'Missions',
    count: 2,
    isLoading: false,
    itemTitle: 'Active',
    itemInfo: 'Software upgrade due',
    apiEndpoint: '/users/count_active',
    loc: {addLink:'/addlocation',link:'/'},
    onClick: () => {}
  }
  expect(shallow(
    <Provider store={store}>
      <DashboardActions {...mockItem} />
    </Provider>
  )).toMatchSnapshot()
})
