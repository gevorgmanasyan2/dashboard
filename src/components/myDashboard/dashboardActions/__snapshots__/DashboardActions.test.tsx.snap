// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Regulatory component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(DashboardActions)
    apiEndpoint="/users/count_active"
    count={2}
    icon={<ForwardRef(RightCircleOutlined) />}
    isLoading={false}
    itemInfo="Software upgrade due"
    itemTitle="Active"
    loc={
      Object {
        "addLink": "/addlocation",
        "link": "/",
      }
    }
    onClick={[Function]}
    title="Missions"
  />
</ContextProvider>
`;
